framework/source/class/qx/io/remote/Rpc.js

```diff
@@ -1072,9 +1102,320 @@ qx.Class.define("qx.io.remote.Rpc",
+            );
+            return reject(ex);
+          }
+
+          // parse response json
+          response = qx.lang.Json.parse(response);
+
+          // check for valid jsonrpc v2 response
+          if (!qx.lang.Type.isArray(response) && !qx.lang.Type.isObject(response)) {
+            ex = new qx.io.remote.exception.Transport(
+              qx.io.remote.RpcError.type.INVALID_DATA,
+              "Invalid jsonrpc data. ",
+              {
+                request : data,
+                response: evt.getContent()
+              }
+            );
+            return reject(ex);
+          }
+
+          // normalize batch and non-batch responses
+          var batch = qx.lang.Type.isArray(response) ? response : [response];
+
+          // handle each response
+          batch.forEach(function(response) {
+            // check for valid jsonrpc v2 response
+            if (!qx.lang.Type.isObject(response)) {
+              ex = new qx.io.remote.exception.Transport(
+                qx.io.remote.RpcError.type.INVALID_DATA,
+                "Invalid jsonrpc result data for "+ response.method,
+                {
+                  request : data,
+                  response: evt.getContent()
+                }
+              );
+              return reject(ex);
+            }
+
+            var ids = [];
+            // analyse individual response
+            var result = response.result;
+            var id = response.id;
+            var method = response.method;
+            var error = response.error;
+
+            // we have the response to our request
+            if (typeof id != "undefined" && typeof method == "undefined") {
+              // we already have a response with the request id
+              if (ids.indexOf(id) !== -1) {
+                ex = new qx.io.remote.exception.Transport(
+                  qx.io.remote.RpcError.type.INVALID_DATA,
+                  "Invalid jsonrpc data: multiple responses with same id.",
+                  {
+                    request : data,
+                    response: response
+                  }
+                );
+                return reject(ex);
+              }
+              ids.push(id);
+              // find original request
+              var originalRequest = this.__requestQueue.find(function (item) {
+                return item.id === id;
+              });
+              // the result id doesn't match any request id
+              if (!originalRequest) {
+                ex = new qx.io.remote.exception.Transport(
+                  qx.io.remote.RpcError.type.INVALID_DATA,
+                  "Invalid jsonrpc data: Unknown request id",
+                  {
+                    request : data,
+                    response: response
+                  }
+                );
+                return reject(ex);
+              }
+              // remove the entry from the queue
+              this.__requestQueue.splice(this.__requestQueue.indexOf(originalRequest), 1);
```

Comment:
This, too, is a slow operation that would be made infinitely faster if `__requestQueue` were a map instead of an array.